### PR TITLE
[Backport 8.1]: Mark command as processed even on processing error 

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

Backports #13206 

Backporting was not that easy, since the module split was done on 8.2 and several classes have been moved etc.

For simplicity I created a separate Test file for the regression test, which already uses the StreamPlatform. Before the StreamProcessorReplayTest used the StreamProcessorRule which is a bit limited.

Furthemore I had to do some more additions to the StreamPlatform to be able to run our test.

## Related issues

closes https://github.com/camunda/zeebe/issues/13101

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: engine/src/test/java/io/camunda/zeebe/engine/processing/streamprocessor/MigratedStreamProcessorReplayTest.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+final class MigratedStreamProcessorReplayTest {
+
+  private static final long TIMEOUT_MILLIS = 2_000L;
+  private static final VerificationWithTimeout TIMEOUT = timeout(TIMEOUT_MILLIS);
+
+  @SuppressWarnings("unused") // injected by the extension
+  private StreamPlatform streamPlatform;
+
+  @RegressionTest("https://github.com/camunda/zeebe/issues/13101")
+  public void shouldNotReplayErrorEventAppliedInSnapshot() throws Exception {
```
Comment (Zelldon) [inline engine/src/test/java/io/camunda/zeebe/engine/processing/streamprocessor/MigratedStreamProcessorReplayTest.java:46]: I extracted this test from the StreamProcessorReplayTest https://github.com/camunda/zeebe/blob/main/stream-platform/src/test/java/io/camunda/zeebe/stream/impl/StreamProcessorReplayTest.java#L269-L334

File: engine/src/test/java/io/camunda/zeebe/engine/util/StreamPlatform.java [hunk 2]
```diff
@@ -52,6 +53,7 @@
 import java.util.concurrent.TimeUnit;
 import java.util.stream.Stream;
 import java.util.stream.StreamSupport;
+import org.mockito.Mockito;
 import org.slf4j.Logger;
 
 public final class StreamPlatform {
```
Comment (Zelldon) [inline engine/src/test/java/io/camunda/zeebe/engine/util/StreamPlatform.java:59]: The changes in this class where necessary in order to make our test work. They are already available on main.

**Review by deepthidevaki — APPROVED** [review]
Thanks :rocket:

**Comment by Zelldon** [comment]
Bors r+

**Pull request merged**